api/v1beta1/cinderapi_types.go

```diff
@@ -28,8 +28,9 @@ type CinderAPITemplate struct {
 
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=0
 	// Replicas - Cinder API Replicas
-	Replicas int32 `json:"replicas"`
+	Replicas *int32 `json:"replicas"`
```

Comment:
As an old school C programmer in the distant past, I'm struggling to understand how this works. I gather the defaulting webhook will zero these fields, yielding the equivalent of a NULL pointer. Then the kubebuilder annotation will result in the replicas field being updated to point (because it's a pointer) to the default value, meaning the pointer is updated. Somehow kubebuilder knows how to set default values by reference AND by value (two different use cases).

Is my understanding anything close to correct?

Reply:
> I gather the defaulting webhook will zero these fields

I'm not sure if the defaulting webhook sets the pointers to the zero-value of `nil` or if it just ignores them completely and thus they would remain `nil` as they would presumably already be.  But anyhow, then when they subsequently reach the kubebuilder logic, that logic indeed knows how to set the default value whether it's by reference or by value.  So your understanding is correct.